src/ast/passes/semantic_analyser.cpp

```diff
@@ -721,8 +722,10 @@ void SemanticAnalyser::visit(Call &call)
           << typestr(arg.type.GetTy());
     }
 
-    size_t max_buffer_size = bpftrace_.config_.get(ConfigKeyInt::max_strlen);
-    size_t buffer_size = max_buffer_size;
+    // Subtract out metadata headroom
+    uint32_t max_buffer_size = bpftrace_.config_.get(ConfigKeyInt::max_strlen) -
```

Comment:
nit just to distinguish these two:
```suggestion
    const uint32_t max_buffer_size = bpftrace_.config_.get(ConfigKeyInt::max_strlen) -
```